src/datasource/utils.ts

```diff
@@ -543,3 +485,125 @@ export function swap<T>(list: T[], n: number, k: number): T[] {
   }
   return newList;
 }
+
+export function bindFunctionDefs(functionDefs, category) {
```

Comment:
Can we also have some unit tests for the functions we moved which don't have any unit test? I think it's a good opportunity to have some.